### PR TITLE
Feature/67 redesign home page

### PR DESCRIPTION
**Pull request by voznovych**

### CODE REVIEW

**Review by eugent482 — APPROVED** [review]

File: InternetHospital.WebApi/Controllers/ArticleController.cs [hunk 1]
```diff
@@ -23,6 +23,14 @@ public ArticleController(IArticleService articleService, IArticleTypeService art
             _articleTypeService = articleTypeService;
         }
 
+        [HttpGet]
+        public IActionResult GetArticles([FromQuery] HomePageArticlesFilteringModel filter)
+        {
+            var articles = _articleService.GetArticlesForHomePage(filter);
+            return Ok(articles);
```
Comment (TREN-D) [inline InternetHospital.WebApi/Controllers/ArticleController.cs]: Is it okay to return ok() with arcticles if it can be null?

**Review by TREN-D — APPROVED** [review]

**Review by edward0802 — APPROVED** [review]

File: InternetHospital.BusinessLogic/Services/ArticleService.cs [hunk 2]
```diff
@@ -293,5 +293,37 @@ private int AddTypesToExistingArticle(IEnumerable<int> typeIds, int articleId)
...
+                    Types = a.Types.Select(t => t.Type.Name).ToList(),
+                    DateOfCreation = a.TimeOfCreation,
+                    ShortDescription = a.ShortDescription,
+                    Author = $"{a.Author.FirstName} {a.Author.SecondName}",
+                    PreviewImageUrls = a.Attachments.Where(att => att.IsOnPreview).Select(att => att.Url).ToList()
+                }).ToList();
+                return result;
+            }
+
+            return null;
```
Comment (OleksandrAntoniuk) [inline InternetHospital.BusinessLogic/Services/ArticleService.cs]: bad practice

File: InternetHospital.WebApi/Controllers/ArticleController.cs [hunk 3]
```diff
@@ -23,6 +23,19 @@ public ArticleController(IArticleService articleService, IArticleTypeService art
             _articleTypeService = articleTypeService;
         }
 
+        [HttpGet]
+        public IActionResult GetArticles([FromQuery] HomePageArticlesFilteringModel filter)
+        {
+            var articles = _articleService.GetArticlesForHomePage(filter);
```
Comment (OleksandrAntoniuk) [inline InternetHospital.WebApi/Controllers/ArticleController.cs]: just Ok(_articleService.GetArticlesForHomePage(filter))

**Review by OleksandrAntoniuk — CHANGES_REQUESTED** [review]

File: InternetHospital.BusinessLogic/Services/ArticleService.cs (hunk 2 above)
Comment (voznovych) [reply InternetHospital.BusinessLogic/Services/ArticleService.cs]: Than better return just empty object?

File: InternetHospital.BusinessLogic/Models/Articles/HomePageArticleModel.cs [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+using System.Text;
+
+namespace InternetHospital.BusinessLogic.Models.Articles
+{
+    public class HomePageArticleModel
+    {
+        public string Title { get; set; }
+        public string ShortDescription { get; set; }
+        public string Author { get; set; }
+        public DateTime DateOfCreation { get; set; }
```
Comment (OleksandrAntoniuk) [inline InternetHospital.BusinessLogic/Models/Articles/HomePageArticleModel.cs]: just CreatedDate

**Review by OleksandrAntoniuk — APPROVED** [review]

**Pull request merged**